dbt/adapters/clickhouse/impl.py

```diff
@@ -412,15 +412,30 @@ def get_model_settings(self, model):
         res = []
         for key in settings:
             res.append(f' {key}={settings[key]}')
-        return '' if len(res) == 0 else 'SETTINGS ' + ', '.join(res) + '\n'
```

Comment:
Let's always add the `-- end_of_sql` marker comment regardless of whether we have actual settings to add?  That way we can check for it in other circumstances where we need to do stuff after the "main" sql.

Reply:
Do you mean to add that at the end of every macro?

Reply:
It would be the same macros where we add `settings` now, just _always_ add `-- end_of_sql` instead of conditionally.

I think the `settings` thing is fairly unique to ClickHouse so I feel like being able to consistently determine "this is the end of regular SQL" from "this might be ClickHouse doing nonstandard stuff" will lead to more straightforward code, even if we don't actually add the settings.  Note that there are so many possible query and table settings that we'll probably end up with them most of the time anyway.

Reply:
Done

Reply:
Opened a new PR in #249